tests/tests/test_basics.py

```diff
@@ -305,7 +307,44 @@ def test(self, caplog):
         # message.
         run_testsuite(self.Mysuite, ["--temp-dir=tmp"], expect_failure=True)
         logs = testsuite_logs(caplog)
-        assert "temp dir 'tmp' does not exist" in logs
+        assert any(
+            re.search("temp dir '.*tmp' does not exist", entry)
+            for entry in logs
+        )
+
+
+class TestNoRandomTestDir:
+    """Check support for --no-random-temp-subdir."""
+
+    class MyDriver(BasicDriver):
+        def run(self, prev, slot):
+            mkdir(self.working_dir())
+            return True
+
+        def analyze(self, prev, slot):
+            self.result.set_status(Status.PASS)
+            self.push_result()
+
+    class Mysuite(Suite):
+        tests_subdir = "simple-tests"
+
+        @property
+        def test_driver_map(self):
+            return {"default": TestNoRandomTestDir.MyDriver}
+
+        default_driver = "default"
+
+    def test(self, tmp_path):
+        run_testsuite(
+            self.Mysuite,
+            [
+                f"--temp-dir={tmp_path}",
+                "--no-random-temp-subdir",
+                "--cleanup-mode=none",
+            ],
+        )
+        assert os.path.isdir(str(tmp_path / "test1"))
```

Comment:
Out of curiosity, what type is tmp_path, and how does this work?

Reply:
See the corresponding doc in the py.test framework: https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture